.github/workflows/setup-e2e-test/action.yaml

```diff
@@ -0,0 +1,48 @@
+name: Setup E2E test template
+description: A composite action to setup e2e tests
+
+inputs:
+  kubernetes-version:
+    required: true
+    description: kubernetes version
+  python-version:
+    required: true
+    description: Python version
```

Comment:
One small thing is that if there are other steps that need some of these versions, for example gang-scheduler-name [here](https://github.com/kubeflow/training-operator/blob/4ea4bdecc6b14aa815a7e5e446d4e41eaf8b3ee6/.github/workflows/integration-tests.yaml#L77), we will need to export them as environments variables to access in subsequent steps using [GITHUB_ENV](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-an-environment-variable). Is there any other way or is this fine?

Reply:
Oh, I see. I guess, we only use scheduler plugins for integration tests.
@kubeflow/wg-training-leads @saileshd1402 do we want to tests our Notebooks with various scheduling plugins as well ?
Or we want to limit the tests that we run with gang-scheduling ?

Reply:
> Should we set the matrix with Kubernetes and Python versions as part of our setup-e2e-test template ?
Right now, we set it in the [integration-tests.yaml](https://github.com/kubeflow/training-operator/blob/4ea4bdecc6b14aa815a7e5e446d4e41eaf8b3ee6/.github/workflows/integration-tests.yaml#L29).
So we can keep it consistent for our E2Es + Notebooks tests.
WDYT @tenzen-y @Electronic-Waste @saileshd1402 ?

I think so. It will be better if we could execute e2e tests with multiple Kubernetes and Python versions.

Reply:
I guess, maybe gang-scheduling could be limited to only `intergration-test.yaml` since it is a bit redundant to test them again in notebook tests.

Reply:
@saileshd1402 Maybe to unblock this PR, we can just use GITHUB_ENV for now and set the gang-scheduler only for integration tests.
For the V2 tests, we can come back to this discussion.

Reply:
I found out that we can't use matrix inside a single composite action. It can only be used in jobs/workflows files. This is because composite action avoids duplication of steps but can't be used to create more jobs like a workflow file. There is also [Reusable Workflows](https://docs.github.com/en/actions/sharing-automations/reusing-workflows#creating-a-reusable-workflow) but that can't be used in this case sine it's spawns separate workflow to run the template, which means that we can't use it to setup environment of current job. Related docs: [matrix strategies](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow#about-matrix-strategies) and [composite actions](https://docs.github.com/en/actions/sharing-automations/creating-actions/about-custom-actions#composite-actions).

Reply:
I see, thanks for checking!